src/libstd/ffi/c_str.rs

```diff
@@ -255,6 +255,34 @@ impl CString {
         Box::into_raw(self.inner) as *mut libc::c_char
     }
 
+    /// Converts the `CString` into a `String` if it contains valid Unicode data.
+    ///
+    /// On failure, ownership of the original `CString` is returned.
+    #[unstable(feature = "convert", reason = "recently added", issue = "27704")]
+    pub fn into_string(self) -> Result<String, CString> {
```

Comment:
Could you suggest what this type should be called? Would the best course of action be to just create  `std::ffi::FromUtf8Error` or name it `CStringFromUtf8Error` or..?